code/modules/clothing/clothing.dm

```diff
@@ -216,6 +216,12 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+/obj/item/clothing/proc/on_mob_life(var/mob/living/M)
```

Comment:
There's already an `OnMobLife` in generic item code, better use this.

Make sure to check whether the user's wearing it.